ecs-agent/tmds/server.go

```diff
@@ -45,7 +45,7 @@ type Config struct {
 	writeTimeout    time.Duration // http server write timeout
 	steadyStateRate float64       // steady request rate limit
 	burstRate       int           // burst request rate limit
-	router          *mux.Router   // router with routes configured
+	handler         http.Handler  // HTTP handler with routes configured
```

Comment:
clarification: is changing from pointer (*mux.Router) to http.Handler intentional?

Reply:
Yeah because `http.Handler` is an `interface` whereas `mux.Router` is a `struct`.